phoenicis-repository/src/main/java/org/phoenicis/repository/DynamicRepositoryLocationLoader.java

```diff
@@ -0,0 +1,20 @@
+package org.phoenicis.repository;
+
+import org.phoenicis.repository.location.RepositoryLocation;
+import org.phoenicis.repository.repositoryTypes.Repository;
+import org.springframework.context.ApplicationContext;
+
+import java.util.List;
+
+public class DynamicRepositoryLocationLoader implements RepositoryLocationLoader {
+    private final RepositoryLocationLoader delegated;
+
+    public DynamicRepositoryLocationLoader(ApplicationContext applicationContext, String beanName) {
```

Comment:
Is there a way to use the `className` instead of the `beanName` here?

Reply:
This is really the name of the bean that we consider here, not the name of the class

Reply:
I know.
I just think it would be nice to use the class name instead of the bean name.
Especially in the config/properties file I believe it to be more intuitive to see the class name than the bean name which is specified inside an on the first view arbitrary java file (i.e. `RepositoryConfiguration`).

Reply:
The thing is that you could have several beans with the same class theoritically. For example something like systemFileSystemLocationLoader and a userHomeLocationLoader could both be FilesystemJsonRepositoryLocationLoader

Reply:
Couldn't we use the fully qualified name (i.e. package + class name)?
This name should be unique too, right?

Reply:
No because we can have two beans with different names but using the same class. They will only differ by the parameters you pass on their respective constructors

Reply:
A bean represents an instance of a given class. 

In most cases, these instance are singletons because they are statically declared (i.e. in a configuration file), but this is not always true. This is why spring introduces the concept of bean name